### DISCUSSION TITLE
How to combine `sanitycheck --build-only` and `sanitycheck --test-only` using hardware-map ?

### DISCUSSION THREAD

**Discussion by erwango** [zephyrproject-rtos/zephyr#28733] [Q&A]
I'm currently running 2 sanitycheck independent commands:
- `sanitycheck -b -N` on a wide list of boards (`-p x -p y -p z...`)
- `sanitycheck -device-testing --hardware-map ../map.yaml` which runs on the devices attached to my bench.

I'd like to take advantage of the first build session to speed up the device testing session using `-test-only`.
Though, from what I see, when using `test-only`, the script will try to run tests on boards x y and z, even if only x is connected to the bench.

Is there a way to limit testing on the boards connected to the bench with `test-only` option ?

**Comment by erwango** [comment]
@nashif, might be a question for you.

**Comment by nashif** [comment]
well, when running `sanitycheck -device-testing` you probably want to limit the testing to specific boards using `-p x -p y -p ...` as well

**Comment by galak** [comment]
While not specific to your issue we just merged support so you can limit the build to only things that are runnable.

**Comment by erwango** [comment]
> well, when running sanitycheck -device-testing you probably want to limit the testing to specific boards using -p x -p y -p ... as well

Actually the set of boards I want to run `--device-testing --test-only` on is a subset of the `--build-only` set.
And this doesn't work well today as script expect that _all boards from `--build-only`_ to be present for testing. 

To be more specific, I build on all STM32 based targets, but I don't have all of them on my bench...

**Comment by erwango** [comment]
I've raised https://github.com/zephyrproject-rtos/zephyr/issues/29166 as it appears there is nothing that can be done to prevent this behavior.
